### ISSUE TITLE
DOC202 false positive with abstract methods

### ISSUE THREAD

**Issue by mhooreman** [astral-sh/ruff#12685]
Hello,

### Context:
- An abstract method is defined
- The docstring of this abstract method has a return statement
- There is (obviously) no implementation in the abstract method ... and so no return statement

### Actual result:
 `DOC202 Docstring should not have a returns section because the function doesn't return anything`

### Expected result:
No error

### Keywords searched in issues list before creating this issue

- DOC202

### Code snippet:

```python
import abc

class A(metaclass=abc.abcmeta):
    @abc.abstractmethod
    def f(self):
        """Lorem ipsum

        Returns
        -------
        dict:
            The values
        """

class C(A):
    def f(self):  # noqa: PLR6301 - this is for demo purpose
        return {}
```

### Remarks:
- Same result with type annotation given in the abstract method definition (e.g. `def f(self) -> dict:`)
- In terms of implementation, a possibility could be to check if the method is decorated as abstract before doing this test

**Comment by charliermarsh** [comment]
Do you still see this in the latest release? We ignore stub functions already, so the example above doesn't raise a violation: https://play.ruff.rs/c6b7a90d-26a2-4a50-9b0a-8d5b0259a85f

**Comment by edgarrmondragon** [comment]
> Do you still see this in the latest release? We ignore stub functions already, so the example above doesn't raise a violation: https://play.ruff.rs/c6b7a90d-26a2-4a50-9b0a-8d5b0259a85f

0.5.7 did seem to fix it for me :)
